Slim/Control/Queries.pm

```diff
@@ -4281,6 +4285,25 @@ sub statusQuery {
 									$data, $tags,
 									'playlist index', $idx, $fast
 								);
+
+						if ( $tags =~ /2/ ) {
+							# build a hash containing the grouping data for each album instance/track in the play queue.
+							my $track = $request->{_results}->{'playlist_loop'}[$count];
+							my $album = $track->{'album_id'};
+							my $tracknum = $track->{'tracknum'};
+							my $albumTrack = $album*10000+$tracknum; # used to detect a new album or a repeat of tracks from the same album
+							my $work = $track->{'work_id'} || '';
+							my $grouping = $track->{'grouping'} || '';
+							my $performance = $track->{'performance'} || '';
+							my $group = "$work##$grouping##$performance";
+							if ( $album != $lastAlbum || $albumTrack < $lastAlbumTrack ) {
+								++$albumNumber;
+							}
+							$lastAlbum = $album;
+							$lastAlbumTrack = $albumTrack;
+							$request->{_results}->{'playlist_loop'}[$count]->{'albumNumber'} = $albumNumber;
```

Comment:
Oooh... don't mess with a request's inner data structures. Store the data elsewhere if needed. But don't modify an object's internal data.